lib/composite_primary_keys/associations/collection_association.rb

```diff
@@ -2,17 +2,24 @@ module CompositePrimaryKeys
   module CollectionAssociation
     def ids_writer(ids)
       primary_key = reflection.association_primary_key
-      pk_type = klass.type_for_attribute(primary_key)
       ids = Array(ids).reject(&:blank?)
-      ids.map! { |i| pk_type.cast(i) }
 
       # CPK-
       if primary_key.is_a?(Array)
+        ids = CompositePrimaryKeys.normalize(ids, primary_key.size)
+        primary_key.each_with_index do |primary_key, i|
+          pk_type = klass.type_for_attribute(primary_key)
+          ids.each { |id| id[i] = pk_type.cast(id[i]) }
```

Comment:
Is this where the string is split by comma? If not, where is that code?

Reply:
It's done [here](https://github.com/composite-primary-keys/composite_primary_keys/pull/574/files#diff-031aba8efafc35d1c9e42feddeef3a74c2acfbd8e66a8b4acb0e92d1333b527aR9):

```ruby
ids = CompositePrimaryKeys.normalize(ids, primary_key.size)
```

It uses [CompositePrimaryKeys#normalize](https://github.com/composite-primary-keys/composite_primary_keys/blob/master/lib/composite_primary_keys/composite_arrays.rb#L22) for parsing, so it supports unescaping commas originally contained in the ids.

But I realized that it's better to use [CompositePrimaryKeys::CompositeKeys#parse](https://github.com/composite-primary-keys/composite_primary_keys/blob/master/lib/composite_primary_keys/composite_arrays.rb#L41) to reduce ambiguity. #normalize's recursive behavior may lead to an unintended result, if a comma is contained in the id.

Reply:
Wouldn't normalize have the same problem if the keys are strings that contain commas? In this example, assume ";" is the separator"

"A key with an , comma; A second key, 5332"

Its not very likely to happen, but I'm sure someone has this. Of course, its probably wouldn't work with CPK anyway since I don't believe its ever been tested.

Reply:
Sorry I don't understand what you're worried about.
Even if ids are String and it contain commas, the comma-separated String is generated by [CompositePrimaryKeys::CompositeKeys#to_s](https://github.com/composite-primary-keys/composite_primary_keys/blob/master/lib/composite_primary_keys/composite_arrays.rb#L54) anyway. It escapes potentially conflicting characters, so there'll be no issue.

Reply:
Good point. I forgot about https://github.com/composite-primary-keys/composite_primary_keys/pull/506.